src/gather_classes.rs

```diff
@@ -270,36 +274,40 @@ fn update_report(
 }
 
 // Converts recursive calls to value.get("calls") to a boxed iterator.
-fn get_calls<'a>(obj: &'a Value) -> Box<dyn Iterator<Item = &Value> + 'a> {
+fn get_calls<'a>(obj: &'a Value) -> Result<Box<dyn Iterator<Item = &Value> + 'a>, anyhow::Error> {
     // TODO (#72) Put debug logging here under a core::option_env flag so it is normally hidden.
     match obj {
-        Value::String(_same) => Box::new(::std::iter::empty()),
+        Value::String(_same) => Ok(Box::new(::std::iter::empty())),
         Value::Array(call_list) => {
```

Comment:
You're going to have some conflict with one of your previous PRs, I think this one should be merged last